ihmc-high-level-behaviors/src/main/java/us/ihmc/behaviors/tools/TrajectoryRecordReplay.java

```diff
@@ -1,276 +1,470 @@
+      IntFunction<TrackerData> trackerDataSupplier = index -> segmentType.getSegmentSide() == null ? vrInputData.get(index).trackerData.get(trackerIndex.get(segmentType)) : vrInputData.get(index).controllerData.get(segmentType.getSegmentSide());
+      packTrackerDataInternal(trackerDataSupplier, poseToPack, angularVelocityToPack, linearVelocityToPack);
    }
 
-   public void setDoneReplay(boolean doneReplaying)
+   private void packTrackerDataInternal(IntFunction<TrackerData> trackerSupplier, FramePose3D poseToPack, FrameVector3D angularVelocityToPack, FrameVector3D linearVelocityToPack)
    {
-      this.doneReplaying = doneReplaying;
+      if (replayInterpolationAlpha == -1.0)
+      {
+         TrackerData trackerData = trackerSupplier.apply(replayIndex);
+         poseToPack.set(trackerData.pose);
+         angularVelocityToPack.set(trackerData.angularVelocity);
+         linearVelocityToPack.set(trackerData.linearVelocity);
+      }
+      else
+      {
+         TrackerData trackerData0 = trackerSupplier.apply(replayIndex - 1);
+         TrackerData trackerData1 = trackerSupplier.apply(replayIndex);
+
+         poseToPack.interpolate(trackerData0.pose, trackerData1.pose, replayInterpolationAlpha);
+         angularVelocityToPack.interpolate(trackerData0.angularVelocity, trackerData1.angularVelocity, replayInterpolationAlpha);
+         linearVelocityToPack.interpolate(trackerData0.linearVelocity, trackerData1.linearVelocity, replayInterpolationAlpha);
+      }
    }
 
-   public String getPath()
+   ////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+   /////////////////////////////////////////////////////////   Helper methods   ///////////////////////////////////////////////////////////
+   ////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+   private static boolean getAButtonPressed(RobotSide robotSide, int index, List<VRInputData> vrInputData)
    {
-      return this.filePath;
+      ControllerData controllerData = vrInputData.get(index).controllerData.get(robotSide);
+      return controllerData.aButtonPressed;
    }
 
-   public void setPath(String filePath)
+   private static boolean getBButtonPressed(RobotSide robotSide, int index, List<VRInputData> vrInputData)
    {
-      this.filePath = filePath;
-      this.reset();
+      ControllerData controllerData = vrInputData.get(index).controllerData.get(robotSide);
+      return controllerData.bButtonPressed;
    }
 
-   public void setPath(String filePath, boolean reset)
+   private static boolean getTriggerPressed(RobotSide robotSide, int index, List<VRInputData> vrInputData)
    {
-      this.filePath = filePath;
-      if (reset)
-         this.reset();
+      ControllerData controllerData = vrInputData.get(index).controllerData.get(robotSide);
+      return controllerData.triggerPressed;
    }
 
```

Comment:
Checks button press over the interval from the previous update to this update, so that buttons aren't missed or replayed multiple times.